gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -203,6 +211,35 @@ var gpii = fluid.registerNamespace("gpii");
         return fullPayload;
     };
 
+    // Transform preferences to final payload and apply the final payload
+    gpii.lifecycleManager.applyPreferences = function (that, preferences, prefsAppliedLocalEvent) {
+        fluid.log("Lifecycle Manager: Received updated preferences ", preferences);
+
+        // Grab current session
+        var session = that.getSession(that.getActiveSessionGpiiKeys());
+
+        // tweak payload to be in preferences set format and update session
+        var contextualPrefs = {};
+        fluid.each(preferences, function (spec, pref) {
+            fluid.set(contextualPrefs, fluid.pathUtil.parseEL(pref), spec.value);
+        });
+
+        // The updated preferences meant to be saved to cloud should be have the applier change fired with source === "PrefsUpdated"
```

Comment:
This is a pretty strange choice of namespace and will not really help anyone who reads it to understand the purpose - since clearly the preferences are always being updated if there is change to "preferences". Could you pick a namespace which more clearly defines what the route of update is. Perhaps something like "userUpdate" or "directUserUpdate"